q2_cutadapt/tests/test_demux.py

```diff
@@ -200,6 +200,27 @@ def test_batch_size_odd_number_of_samples(self):
         # obs_untrimmed should be empty, since everything matched
         self.assert_untrimmed_results(b'', obs_untrimmed_art)
 
+    def test_min_length(self):
+        metadata = CategoricalMetadataColumn(
+            # The third barcode is meant to completely remove the only GGGG
+            # coded sequence
+            pd.Series(['AAAA','CCCC','GGGGACGTACGT'], name='Barcode',
+                      index=pd.Index(['sample_a', 'sample_b', 'sample_c'],
+                      name='id')))
+
+        with redirected_stdio(stderr=os.devnull):
+            obs_demuxed_art, obs_untrimmed_art = \
+                self.demux_single_fn(self.muxed_sequences, metadata,
+                                     batch_size=2)
+
+        obs = obs_demuxed_art.view(SingleLanePerSampleSingleEndFastqDirFmt)
+        num_files = 0
+        for _, obs_fp in obs.sequences.iter_views(FastqGzFormat):
+            num_files += 1
+        # Since the only GGGG coded sequence was removed, only AAAA and CCCC
+        # files should exist leaving two files
+        self.assertEqual(num_files, 2)
```

Comment:
Maybe instead of checking the number of files, we could check _which samples_ are present in the results here. One option is `zip` this up with a list of filenames that you expect to see there. Let us know if you need a hand!

Reply:
Do you want me to verify specifically which samples are left or just that files were only created for AAAA and CCCC?